tests/integration/test_conftest/test_conftest.py

```diff
@@ -0,0 +1,171 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+"""
+The aim of this module is to test db isolation between single tests.
+The `app` fixture does not provide db isolation between single tests: this
+means that if a test adds an object to the db, the same obj is found
+in the next tests.
+The `isolated_app` fixture instead provides db isolation between single tests.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from flask_sqlalchemy import models_committed
+
+from invenio_db import db
+from invenio_oauthclient.models import User
+from invenio_pidstore.models import PersistentIdentifier
+
+
+@pytest.fixture(scope='module')
+def pids_count(_app):
+    # Count the # PIDs currently loaded in the demosite.
+    return PersistentIdentifier.query.count()
+
+
+@pytest.fixture(scope='module')
+def users_count(_app):
+    # Count the # User currently loaded in the demosite.
+    return User.query.count()
+
+
+# Test `app` fixture.
+
+
+def test_app_fixture_lacks_db_isolation_step1(pids_count, app):
+    assert PersistentIdentifier.query.count() == pids_count
+
+    PersistentIdentifier.create(
+        pid_type='type1',
+        pid_value='value1',
+    )
+    # The #PIDs must have incremented.
+    assert PersistentIdentifier.query.count() == pids_count + 1
+
+
+def test_app_fixture_lacks_db_isolation_step2(pids_count, app):
+    assert PersistentIdentifier.query.count() == pids_count + 1
+    # Force the cleanup.
+    PersistentIdentifier.get(
+        pid_type='type1',
+        pid_value='value1',
+    ).delete()
+
+
+# Test `isolated_app` fixture.
+
+
+def test_isolated_app_db_isolation_step1(pids_count, isolated_app):
+    assert PersistentIdentifier.query.count() == pids_count
+    pid = PersistentIdentifier.create(pid_type='type1', pid_value='value1')
+    assert PersistentIdentifier.query.get(pid.id)
+    assert PersistentIdentifier.query.count() == pids_count + 1
+
+
+def test_isolated_app_db_isolation_step2(pids_count, isolated_app):
+    assert PersistentIdentifier.query.count() == pids_count
+    pid = PersistentIdentifier.create(pid_type='type1', pid_value='value1')
+    assert PersistentIdentifier.query.get(pid.id)
+    assert PersistentIdentifier.query.count() == pids_count + 1
+
+
+def test_isolated_app_session_commit(users_count, isolated_app):
+    assert User.query.count() == users_count
+    user = User()
+    db.session.add(user)
+    db.session.commit()
+
+    user.email = 'foo@bar.com'
+    db.session.add(user)
```

Comment:
what's the point of this, if it gets overwritten right after?

Reply:
The point of this block is to ensure `session.merge()` works with `isolated_app`. So I add an user to the session in this line, modify the user in the next line and then merge it to the session.

Reply:
you can do a `merge` without an `add` first, that's why I'm asking.

Reply:
discussing IRL, there should be a `db.session.commit()` here.

Reply:
done